src/clean_air/data/data_subset.py

```diff
@@ -114,16 +112,32 @@ def extract_box(self, box, crs=None):
 
         return cube
 
-    def extract_track(self, track, crs=None):
+    def extract_track(self, start=None, end=None):
         """
         Extract a track
 
         Arguments:
-            track: timeseries of the track, as a dataframe
+            start (datetime.time or str): initial time filter limit
+            end (datetime.time or str): end time filter limit
         """
         cube = self._load_cube()
 
-        return util.cubes.extract_series(cube, track)
+        if start and end:
+            timerange = iris.Constraint(time=lambda cell: start <= cell.point < end)
+            cube = cube.extract(timerange)
+            if cube is None:
+                raise ValueError('Empty cube, likely due to time bounds being out of range')
+
+        # remove all coords except for time DimCoord
+        for coord in cube.aux_coords:
+            cube.remove_coord(coord)
+        if len(cube.dim_coords) != 1:
+            print('Found extra dimensions, attempting to remove...')
+            for coord in cube.dim_coords:
+                if coord.standard_name != 'time':
+                    cube.remove_coord(coord)
+
+        return cube
```

Comment:
This work is beautiful.  I feel much happier with this than the previous incarnation.  Very nice.

Reply:
I've adjusted the code so it now works with just one or the other, with accompanying tests in data subset's unit tests.